### ISSUE TITLE
VcpkgInstalledDir MSBuild property is no longer defined

### ISSUE THREAD

**Issue by andy-schmidt** [microsoft/vcpkg#20019]
**Describe the bug**
The `VcpkgInstalledDir` [manifest mode MSBuild property](https://github.com/microsoft/vcpkg/blob/master/docs/users/manifests.md#msbuild-integration) is no longer being defined, and is instead left empty. It appears that [this commit](https://github.com/microsoft/vcpkg/pull/19767/commits/ccb973a8b4c3d9136ae9e69376dbf8e138c20ff0) removed the sections of `vcpkg.targets` containing the property definition for `VcpkgInstalledDir`.

**Environment**
- OS: Windows 10
- Compiler: Visual Studio Enterprise 2019 16.10.3

**To Reproduce**

1. Update your local copy of `master` of this repo to include the commit mentioned above.
2. Create a Visual Studio project that uses vcpkg manifest mode and the port `gtest` and triplet `x64-windows`
3. Add the following property to the `<Link></Link>` section for `Debug|x64`: `<AdditionalDependencies>$(VcpkgInstalledDir)$(VcpkgTriplet)\debug\lib\manual-link\gmock_maind.lib;%(AdditionalDependencies)</AdditionalDependencies>`
4. Write a simple gtest: `#include <gtest/gtest.h> TEST(quick_test, this_should_succeed) { EXPECT_TRUE(true); }`
5. Build the project

**Expected behavior**
The build to successfully build a binary executable linking in the default `main` program entry point for `gtest` (in this case with mocking framework)

**Failure logs** 
The link step fails with
`LINK : fatal error LNK1181: cannot open input file 'x64-windows-static\debug\lib\manual-link\gmock_maind.lib'`

**Additional context**
If you checkout the [parent commit](https://github.com/microsoft/vcpkg/commit/7734e8b89d6d511a95ac27f8d8eed913535b592d) of the commit in question above in your local repo and try to build again, the project will successfully link.

We are using a [private vcpkg registry](https://devblogs.microsoft.com/cppblog/how-to-start-using-registries-with-vcpkg/)  in manifest mode for an internal Windows C++ library and leaving the MSBuild property `VcpkgInstalledDir` undefined has broken our build.

**Comment by PhoebeHui** [comment]
>Create a Visual Studio project that uses vcpkg manifest mode and the port gtest and triplet x64-windows

>LINK : fatal error LNK1181: cannot open input file 'x64-windows-static\debug\lib\manual-link\gmock_maind.lib'

@andy-schmidt, It looks you are using x64-windows triplet, but the failures logs says 'x64-windows-static', could you double confirm the project setting?

I have double confirmed that both x64-windows and x64-windows-static works correctly with classic and manifest mode for me.

**Comment by autoantwort** [comment]
Yes `VcpkgInstalledDir` is no longer defined. I also recognized this. I have to add a vcpkg lib in librarian, but there vcpkg is not added to the library search path, so I have to add `$(_ZVcpkgCurrentInstalledDir)$(_ZVcpkgConfigSubdir)lib` which doesn't feel right. One could leave it there is we use `<Lib>` instead of `<Link>` here:
https://github.com/microsoft/vcpkg/blob/71422c627264daedcbcd46f01f1ed0dcd8460f1b/scripts/buildsystems/msbuild/vcpkg.targets#L86-L89
@andy-schmidt There you see that `manual-link` link is already added to the library search path, so it should be enough to use `gmock_maind.lib;%(AdditionalDependencies)`, so you don't need `$(VcpkgInstalledDir)`.  
Ping @BillyONeal

**Comment by BillyONeal** [comment]
> Yes `VcpkgInstalledDir` is no longer defined. I also recognized this.

To be clear, `VcpkgInstalledDir` was not defined on non-VS2015 before, and it being defined on VS2015 is a bug/limitation of how integration works there.

>I have to add a vcpkg lib in librarian, but there vcpkg is not added to the library search path

It should be being added to the library search path: 

https://github.com/microsoft/vcpkg/blob/642d1ccb276ce13d8a8521b1ef2856f5f7bc9cdd/scripts/buildsystems/msbuild/vcpkg.targets#L88

If it were not all our integration / applocal deployment etc. stuff would be broken.

> @andy-schmidt There you see that `manual-link` link is already added to the library search path, so it should be enough to use `gmock_maind.lib;%(AdditionalDependencies)`, so you don't need `$(VcpkgInstalledDir)`.
> Ping @BillyONeal

That's correct, you should only need to name the `.lib`, not the full path.

**Comment by autoantwort** [comment]
> It should be being added to the library search path

@BillyONeal It is only added to the [Link Task](https://docs.microsoft.com/en-us/visualstudio/msbuild/link-task?view=vs-2019), but not to the [Lib Task](https://docs.microsoft.com/en-us/visualstudio/msbuild/lib-task?view=vs-2019)
